dataactbroker/scripts/update_transaction_agency_codes.py

```diff
@@ -0,0 +1,154 @@
+                CASE WHEN sub.is_frec
+                    THEN (SELECT frec_code from frec WHERE frec.frec_id = sub.frec_id)
+                    ELSE (SELECT cgac_code from cgac where cgac.cgac_id = sub.cgac_id) END agency_code
+            FROM sub_tier_agency sub
+            INNER JOIN cgac ON cgac.cgac_id = sub.cgac_id
+            INNER JOIN frec ON frec.frec_id = sub.frec_id ) agency
+        WHERE {condition_sql}
+            AND {table_name}.{agency_type}_sub_tier_agency_c{suffix} = agency.sub_tier_agency_code;
+        """.format(table_name=table, agency_type=agency_type, condition_sql=condition_sql, suffix=suffix)
+    )
+    sess.commit()
+
+    final_row_count = 0 if args.subtier_code else get_row_count(condition_sql, table, sess)
+    print_report(row_count, final_row_count, agency_type, True)
+
+    if args.missing_agency:
+        logger.info("{} agency code 999 remaining: {} rows ".format(agency_type.title(), final_row_count))
+
+
+def set_update_condition(agency_type, table, suffix, sess, subtier_code=None):
+    """ Changes the condition on which to update based on the type of update (999 vs subtier code).
+
+        Args:
+            agency_type: string indicating what agency type we're checking, either 'funding' or 'awarding'
+            table: string indicating which table to update, either 'detached_award_procurement' or
+                'published_award_financial_assistance'
+            suffix: string to add an 'o' to a column name if it's a funding agency
+            sess: the database connection
+            subtier_code: the sub tier code to update if one is provided, otherwise None
+
+        Returns:
+            The row count for the number of rows that match the query and "WHERE" condition created
+    """
+    if subtier_code:
+        sql_statement = "{}.{}_sub_tier_agency_c{} = '{}' ".format(table, agency_type, suffix, str(subtier_code))
+    else:
+        sql_statement = "{}.{}_agency_code = '999' ".format(table, agency_type)
+
+    row_count = get_row_count(sql_statement, table, sess)
+    print_report(row_count, 0, agency_type)
+
+    return row_count, sql_statement
+
+
+def get_row_count(sql_statement, table, sess):
+    """ Runs a SQL query to get the count of transaction rows based on the type of update. SQL Statement will either be
+        agency_code = '999' or sub_tier_agency_code = 'XXXX' for awarding and funding.
+
+        Args:
+            sql_statement: string containing the WHERE statement for the query
+            table: string indicating which table to count from, either 'detached_award_procurement' or
+                'published_award_financial_assistance'
+            sess: the database connection
+
+        Returns:
+            The count of how many rows in the provided table match the provided WHERE statement
+    """
+    rows = sess.execute("select count(*) from " + table + " where " + sql_statement + ";")
```

Comment:
(optional) Can these be capitalized "SELECT COUNT(*) FROM [table] WHERE [sql_statement];"